codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/StructureGenerator.java

```diff
@@ -92,7 +93,7 @@ public void renderStructure(Runnable runnable, boolean isInputStructure) {
         writer.addImport("fmt");
         Symbol symbol = symbolProvider.toSymbol(shape);
         writer.openBlock("type $L struct {", symbol.getName());
-
+        ValidationGenerator validationGenerator = new ValidationGenerator(model, symbolProvider, writer);
```

Comment:
This still does the same thing as previous code. You'd need to move it to the constructor to gain any optimization.

```
public StructureGenerator(
            final GenerationContext context,
            GoWriter writer,
            StructureShape shape) {
        this.context = context;
        this.model = context.model();
        this.symbolProvider = context.symbolProvider();
        this.writer = writer;
        this.shape = shape;
        this.validationGenerator = new ValidationGenerator(model, symbolProvider, writer)
    }
```

Reply:
Done